sections/sprint-3/goal-position-generation.tex

```diff
@@ -0,0 +1,25 @@
+\section{Goal generation}\label{subsec:goalrefactoring}
+The first implementation of goal generation was done entirely in Unity.
+This means that the goal generation and detection would have been handled on the client-side.
+The idea was to limit the amount of data that needed to be sent across the network.
+But when it was decided that new goal zones should be generated in random locations after a goal had been scored, generating it on the client-side could cause the game state to be different between different players.
+Either by some players not receiving all the necessary ball positions to observe a goal or by generating goal zones in different places.
+One way to solve this was by giving all clients the same seed at the start of the match to generate random positions based of so that the goal zones would be in the same positions across all clients.
```

Comment:
```suggestion
One way to solve this was by giving all clients the same seed at the start of the match to generate random positions so that the goal zones would be in the same positions across all clients.
```